### PR TITLE
SC-5473 - lernstore test automation

### PR DESCRIPTION
**Pull request by yuliashikhareva**
# Description

## Links to Tickets or other pull requests
https://github.com/hpi-schul-cloud/nuxt-client/pull/1370

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/pages/LRNSTRLernStorePage.js [hunk 1]
```diff
@@ -1 +1,117 @@
...
+    
+    },
+    rightNumberOfFoundContentDisplayed: async function(request) {
+        let selectorTotalNum = await driver.$(selectors.numberOfContentOnGUI);
+        let displayedString= await selectorTotalNum.getText();
+        let displayedNum = await stringHelpers.cutStringAfterSymbol(displayedString, ' ');
+        let apiResponse = await apiHelpers.getLernstoreMaterialsAfterRequest(request);
+        await expect(`${apiResponse}`).to.equal(displayedNum);
+    },
+    clickOnContentCard: async function(request) {
```
Comment (alstanki) [inline page-objects/pages/LRNSTRLernStorePage.js:63]: The name of this method is not equal to what it does. If you need it, split it up into smaller methods

File: page-objects/pages/LRNSTRLernStorePage.js [hunk 2]
```diff
@@ -1 +1,117 @@
...
+    },
+    wasRedirectedToContentPage: async function() {
+        await waitHelpers.waitUntilUrlContains(url);
+    },
+    insertSearchRequest: async function(content) {
+        await waitHelpers.waitAndSetValue(selectors.searchField, content);
+        await waitHelpers.waitAndClick(selectors.searchIcon);
+    
+    },
+    rightNumberOfFoundContentDisplayed: async function(request) {
```
Comment (alstanki) [inline page-objects/pages/LRNSTRLernStorePage.js:56]: numberOfDisplayedContentIsEqual (numberOfContents)

File: page-objects/pages/LRNSTRLernStorePage.js [hunk 3]
```diff
@@ -1 +1,117 @@
...
+        let obj = await driver.$(selectors.courseSelector+' .multiselect__content-wrapper > ul > li:nth-child('+courseIndex+') > span');
+        await obj.click();
+
+    },
+    getIndexOfCourseOrTopicInDropdown: async function(container, elementToSearch) {
+        let selector = (container + ' ' +selectors.selectorsNameOfCourses.selectorCourseTitles);
+        try{
+            const courseList = await Promise.all( 
+                await driver.$$((selector)).map(
+                    async (element) => await element.getText()
```
Comment (alstanki) [inline page-objects/pages/LRNSTRLernStorePage.js:89]: This should be a separate method getListOfCoursesOrTopicInDropdown

File: page-objects/pages/coursePages/CRSSAddEditTopicPage.js [hunk 4]
```diff
@@ -34,7 +40,16 @@ module.exports = {
 	    let geoIDSelector = await idContainer.$(".form-control");
 	    await geoIDSelector.setValue(geogebraID);
 	    await driver.pause(500);
-    },
+	},
+	addLernstoreMaterial: async function(name) {
+		await waitHelpers.waitAndClick(selectors.lernstoreBtn);
+		let idContainer = await driver.$("#content-blocks");
```
Comment (alstanki) [inline page-objects/pages/coursePages/CRSSAddEditTopicPage.js:46]: selectors should be separated above the module.export

File: step_definitions/lernstore-steps.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+});
+Then(/^the teacher must be redirected to content page$/, async function () {
+    await lernstorePage.wasRedirectedToContentPage();
+
+});
+When(/^the teacher searches for content (.*)$/, async function (searchRequest) {
+    await lernstorePage.insertSearchRequest(searchRequest);
+
+});
+Then(/^the right number of materials (.*) is displayed$/, async function (searchRequest) {
```
Comment (alstanki) [inline step_definitions/lernstore-steps.js:23]: remove all "the". Please create step definition as simply as possible

**Review by alstanki — CHANGES_REQUESTED** [review]

**Comment by yuliashikhareva** [comment]
that PR has status "work in progress", as soon as it is done, I will remove the label and it will be available for reviews.

**Pull request closed without merging**